File: example/fastlane/files/FlutterChannel.kt


import android.content.Context
import android.content.pm.ApplicationInfo
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.plugin.common.PluginRegistry.Registrar

/// flutter 調用本地的方法名稱
class FlutterCall {
    companion object {
        /// 取得渠道名
        const val getFlavor = "getFlavor"

        /// 取得應用名
        const val getAppName = "getAppName"
    }
}

class FlutterChannel private constructor() : MethodCallHandler {

    var context: Context? = null

    companion object {

        @JvmStatic
        fun registerWith(registrar: Registrar) {
            val channel = MethodChannel(registrar.messenger(), "project_channel")
            val ins = FlutterChannel()
            ins.context = registrar.context().applicationContext
            channel.setMethodCallHandler(ins)
        }
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            FlutterCall.getFlavor -> {
                result.success(BuildConfig.PLATFORM_CODE)
            }
            FlutterCall.getAppName -> {
                result.success(getAppName())
            }
            else -> {
                result.notImplemented()
            }
        }
    }

    private fun getAppName(): String {
        val appInfo: ApplicationInfo = context!!.applicationInfo
        val stringsId = appInfo.labelRes
        return if (stringsId == 0) appInfo.nonLocalizedLabel.toString()
        else context!!.getString(stringsId)
    }
}

File: example/ios/Runner/flutter/FlutterChannel.swift
import Foundation
import Flutter

class FlutterCall {
    static let getFlavor = "getFlavor"
    static let getAppName = "getAppName"
}

class FlutterChannel {
    
    public var infos: [String : Any] {
        return Bundle.main.infoDictionary!
    }
    
    ///多國語言名稱, 若沒有設置, 則使用 bundleName
    public var name: String {
        return infos["CFBundleDisplayName"] as? String ?? bundleName
    }
    
    ///bundle名稱
    public var bundleName: String {
        return infos["CFBundleName"] as? String ?? ""
    }
    
    private init() {}
    
    static func register(with controller: FlutterViewController) {
        let channel = FlutterMethodChannel(name: "project_channel", binaryMessenger: controller.binaryMessenger)
        let ins = FlutterChannel()
        channel.setMethodCallHandler(ins.onMethodCall)
    }
    
    func onMethodCall(call: FlutterMethodCall, result: FlutterResult) {
        switch call.method {
        case FlutterCall.getFlavor:
            /// ios 沒有渠道, 直接回傳空字串
            result("")
        case FlutterCall.getAppName:
            result(name)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
    
}
